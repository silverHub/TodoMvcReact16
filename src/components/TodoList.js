import './TodoList.css';
import React from 'react';
import PropTypes from 'prop-types';
import TodoListItem from './TodoListItem';

function TodoList(props) {

	return (
		<ul className="todo-list">
			{
				props.todos.map(todo => <TodoListItem key={todo.id}
													  todo={todo}
													  removeTodo={props.removeTodo}
													  toggleTodo={props.toggleTodo}
													  editTodo={props.editTodo}/>)
			}
		</ul>

	);
}

TodoList.propTypes = {
	todos: PropTypes.array,
	removeTodo: PropTypes.func,
	toggleTodo: PropTypes.func
};
TodoList.defaultProps={
	todos: []
};

export default TodoList;
