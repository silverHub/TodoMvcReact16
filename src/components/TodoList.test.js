import React from 'react';
import TodoList from './TodoList';
import { shallow } from 'enzyme';
jest.mock('./TodoListItem', () => 'TodoListItem');

describe('testing TodoList', () => {

	test('if renders if not todos', () => {
		expect(shallow(<TodoList />)).toMatchSnapshot();
	});

	test('if renders if not todos', () => {
		const todos = [
			{ id : 1},
			{ id : 2},
			{ id : 3}
		];
		expect(shallow(<TodoList todos={todos} />)).toMatchSnapshot();
	});


});
