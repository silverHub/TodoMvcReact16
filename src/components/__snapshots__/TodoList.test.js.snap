// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing TodoList if renders if not todos 1`] = `
<ul
  className="todo-list"
/>
`;

exports[`testing TodoList if renders if not todos 2`] = `
<ul
  className="todo-list"
>
  <TodoListItem
    key="1"
    todo={
      Object {
        "id": 1,
      }
    }
  />
  <TodoListItem
    key="2"
    todo={
      Object {
        "id": 2,
      }
    }
  />
  <TodoListItem
    key="3"
    todo={
      Object {
        "id": 3,
      }
    }
  />
</ul>
`;
